### PR TITLE
feat(gh-99): set name during extraction

### LINKED ISSUE #99: Set name during extraction
In `@depscloud/api v0.1.14`, we added the intermediary `name` field to aid in the transition to v1beta. Extraction needs to set this information appropriately based on the raw name in addition to the extraction we're doing for the organization / module pairing.

### PR DESCRIPTION
**Pull request by shinnlok**
Fixes #99

Couple notes:

- I wasn't sure if the names/separators should be standardized.
- Api/schema wasn't updated to include name so I think I made all the changes possible

Let me know.

Thanks!

### CODE REVIEW

File: extractor/src/extractors/BuildGradleExtractor.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ export default class BuildGradleExtractor implements Extractor {
                     module: parentModule,
                     versionConstraint: parentVersion,
                     scopes: [ "parent" ],
+                    name: [ parentOrganization, parentModule ].join(":"),
```
Comment (mjpitz) [inline extractor/src/extractors/BuildGradleExtractor.ts:58]: +1. I wasn't specific about this, but this is the joining semantic I was expecting.

Comment (shinnlok) [reply extractor/src/extractors/BuildGradleExtractor.ts]: Updated!

**Review by mjpitz — CHANGES_REQUESTED** [review]
Can you revert your last commit? What you had before was better because it used formats familiar for each.

**Comment by mjpitz** [comment]
Sorry. I started a review earlier and never got completely through. I see how my last comment may have been misinterpreted.

**Comment by shinnlok** [comment]
Oops my bad. Dropped the commit.

File: extractor/src/extractors/PomXmlExtractor.ts [hunk 2]
```diff
@@ -40,6 +40,7 @@ export default class PomXmlExtractor implements Extractor {
                 module: parentArtifactId,
                 versionConstraint: parentVersion,
                 scopes: [ "parent" ],
+                name: [parentGroupId, parentArtifactId].join(";"),
```
Comment (mjpitz) [inline extractor/src/extractors/PomXmlExtractor.ts:43]: Same here. Should be `:`

File: extractor/src/extractors/IvyXmlExtractor.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ export default class IvyXmlExtractor implements Extractor {
                 scopes = confs.split(";");
             }
 
-            dependencies.push({ organization, module, versionConstraint, scopes });
+            dependencies.push({ organization, module, versionConstraint, scopes, name: [ organization , module ].join(";")});
```
[the remark this answers is not in the document]
Comment (shinnlok) [reply extractor/src/extractors/IvyXmlExtractor.ts]: Done!

File: extractor/src/extractors/PomXmlExtractor.ts (hunk 2 above)
Comment (shinnlok) [reply extractor/src/extractors/PomXmlExtractor.ts]: Done!

**Comment by mjpitz** [comment]
Thanks @shinnlok for your pull request!

**Comment by mjpitz** [comment]
@shinnlok anychance you'd be interested in working on the client-side version of this? https://github.com/depscloud/depscloud/issues/100

It's a stepping stone toward v1beta and would drastically simplify our user experience too. It's very similar to this change, but in go and on the client.

**Comment by shinnlok** [comment]
@mjpitz I would love to!

I'm still figuring my way around the codebase so if you can point me in the right direction..

Regarding my comment in the pr description, does schema not have to be updated before then?

**Comment by mjpitz** [comment]
Nope, we can fake it. I'll explain on the ticket.

**Comment by mjpitz** [comment]
I left a comment on the ticket. Let me know if you have any futher questions there!

**Pull request merged**